app/presenters/matrix_box_presenter.rb

```diff
@@ -128,8 +135,9 @@ def user_to_presenter(user)
     self.where = user.location if user.location
     return unless user.image_id
 
+    # Not user.images because that's every image they've uploaded
     self.image_data = {
-      image: user.image_id,
+      images: [Image.find(user.image_id)],
```

Comment:
This is a big problem.  This needs to be passed in somehow or it will result in a query for each matrix box.

Reply:
Changed

Reply:
I'm not sure I understand why that is the case, though. Or if that is the case. 
This code only executes in the case of a User matrix_box, and ...hmmm. It does seem the ImagePresenter avoids the lookup.

In `ImagePresenter`, there's this:
```ruby
    # Sometimes it's prohibitive to do the extra join to images table,
    # so we only have image_id. It's still possible to render the image with
    # nothing but the image_id. (But not votes, original name, etc.)
    image, image_id = image.is_a?(Image) ? [image, image.id] : [nil, image]
```
But upstream of both these presenters, aren't we already eager-loading the Image associations for whatever model?

Reply:
`Image.find(...)` is always going to do a query, no matter how many images have been eager-loaded elsewhere.

Reply:
by the way, will Bullet tell you if you're over-eager-loading?

I feel like I'm just chucking associations in there willy-nilly and i'd feel better about it if it warned me when it's oversalted.

Reply:
It can in some cases.  I tried to enable it throughout all the tests, but it is too sensitive.  I think it is turned on for all of the integration tests and a handful of others.  But not every test.

I found the best thing to do was (when working on things that were likely to have an effect) watch the development logs.

Reply:
Roger. I'm finding one maybe un-optimized thing left on obs index that wasn't caught by Bullet:

Are these image votes, or naming votes? Checking the obs index controller...
```
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516707) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (300.3ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516706) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (297.3ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516705) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516705) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (295.2ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516704) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516704) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (298.6ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516703) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516703) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (300.3ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516702) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516702) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (299.6ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516701) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516701) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (299.7ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516700) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516700) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (297.8ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516699) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (299.6ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516698) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516698) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (296.9ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516697) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516697) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (300.4ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516696) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516696) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (296.7ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516695) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516695) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (304.4ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516694) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516694) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
etc...
```

Reply:
Nope, adding `:votes` didn't affect it.  I did:
```ruby
# app/controllers/observations_controller/index.rb line 192:

    def define_index_args(query, args)
      args = { controller: "/observations",
               action: :index,
               matrix: true,
               include: [:name, :location, :user, :rss_log, **:votes,**
                         { images: [:image_votes, :license, :projects, :user] },
                         { thumb_image: :image_votes }] }.merge(args)
```

Reply:
Yep, there's something funky going on. 
- Why is the obs index even loading `project_images`?
- Immediately after those is a second batch of `images` and `image_votes`.

The reason it's checking `projects` is that Project now participates in defining `can_edit?` for an image. This is really far afield from the business of the Obs index; maybe i'll try maybe eliminating the image captions here. Oh, nope. Captions are used in the lightbox and encoded into every image as a data attribute! Argh. (or rather "Kwargh!" as Rubocop prefers!)

```
          /vagrant/mushroom-observer/app/models/project.rb:115:in `can_edit?'
          /vagrant/mushroom-observer/app/models/image.rb:919:in `can_edit?'
          /vagrant/mushroom-observer/app/controllers/application_controller.rb:472:in `editable_by_user?'
          /vagrant/mushroom-observer/app/controllers/application_controller.rb:464:in `correct_user_for_object?'
          /vagrant/mushroom-observer/app/controllers/application_controller.rb:459:in `check_permission'
          /vagrant/mushroom-observer/app/controllers/application_controller.rb:462:in `check_permission'
          /vagrant/mushroom-observer/app/helpers/image_helper.rb:75:in `show_original_name?'
          /vagrant/mushroom-observer/app/helpers/image_helper.rb:67:in `image_owner_original_name'
          /vagrant/mushroom-observer/app/helpers/image_helper.rb:56:in `image_info'
          /vagrant/mushroom-observer/app/helpers/carousel_helper.rb:73:in `carousel_caption'
```
```
QUERY starting: "\n      SELECT DISTINCT observations.id\n      FROM `observations` \n      ORDER BY observations.log_updated_at DESC, observations.id DESC\n"
  Observation Load (1.9ms)  SELECT `observations`.* FROM `observations` WHERE `observations`.`id` IN (516551, 516707, 516706, 516705, 516704, 516703, 516702, 516701, 516700, 516699, 516698, 516697, 516696, 516695, 516694, 516693, 516692, 516691, 516690, 516689, 516688, 516687, 516686, 516685)
  Name Load (3.0ms)  SELECT `names`.* FROM `names` WHERE `names`.`id` IN (41809, 112489, 112180, 112490, 112491, 112492, 112493, 112494, 112495, 112496, 112497, 112498, 112499, 112500, 112501, 112502, 112503, 112504)
  Location Load (1.1ms)  SELECT `locations`.* FROM `locations` WHERE `locations`.`id` IN (12848, 6042, 27749, 27747, 3097)
  User Load (1.8ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` IN (8496, 455)
  RssLog Load (0.8ms)  SELECT `rss_logs`.* FROM `rss_logs` WHERE `rss_logs`.`id` IN (628633, 628808, 628809, 628810, 628811, 628813, 628815, 628816, 628818, 628820, 628822, 628823, 628825, 628827, 628828, 628830, 628831, 628833, 628835, 628837, 628839, 628841, 628843, 628845)
  Naming Load (5.1ms)  SELECT `namings`.* FROM `namings` WHERE `namings`.`observation_id` IN (516551, 516685, 516686, 516687, 516688, 516689, 516690, 516691, 516692, 516693, 516694, 516695, 516696, 516697, 516698, 516699, 516700, 516701, 516702, 516703, 516704, 516705, 516706, 516707)
  Vote Load (465.4ms)  SELECT `votes`.* FROM `votes` WHERE `votes`.`naming_id` IN (710892, 711066, 711043, 711044, 711045, 711046, 711047, 711048, 711049, 711050, 711051, 711052, 711053, 711054, 711055, 711056, 711057, 711058, 711059, 711060, 711061, 711062, 711063, 711064, 711065)
  ObservationImage Load (0.6ms)  SELECT `observation_images`.* FROM `observation_images` WHERE `observation_images`.`observation_id` IN (516551, 516685, 516686, 516687, 516688, 516689, 516690, 516691, 516692, 516693, 516694, 516695, 516696, 516697, 516698, 516699, 516700, 516701, 516702, 516703, 516704, 516705, 516706, 516707)
  Image Load (0.6ms)  SELECT `images`.* FROM `images` WHERE `images`.`id` IN (1550397, 1550664, 1550665, 1550666, 1550667, 1550668, 1550669, 1550670, 1550671, 1550672, 1550673, 1550674, 1550675, 1550676, 1550677, 1550678, 1550679, 1550680, 1550681, 1550682, 1550683, 1550684, 1550685, 1550686, 1550687, 1550688, 1550689, 1550690, 1550691, 1550692, 1550693, 1550694, 1550695, 1550696, 1550697, 1550698, 1550699, 1550700, 1550701, 1550702, 1550703, 1550704, 1550705, 1550706, 1550707)
  ImageVote Load (0.7ms)  SELECT `image_votes`.* FROM `image_votes` WHERE `image_votes`.`image_id` IN (1550397, 1550664, 1550665, 1550666, 1550667, 1550668, 1550669, 1550670, 1550671, 1550672, 1550673, 1550674, 1550675, 1550676, 1550677, 1550678, 1550679, 1550680, 1550681, 1550682, 1550683, 1550684, 1550685, 1550686, 1550687, 1550688, 1550689, 1550690, 1550691, 1550692, 1550693, 1550694, 1550695, 1550696, 1550697, 1550698, 1550699, 1550700, 1550701, 1550702, 1550703, 1550704, 1550705, 1550706, 1550707)
  License Load (0.4ms)  SELECT `licenses`.* FROM `licenses` WHERE `licenses`.`id` IN (3, 2)
  ProjectImage Load (23.2ms)  SELECT `project_images`.* FROM `project_images` WHERE `project_images`.`image_id` IN (1550397, 1550664, 1550665, 1550666, 1550667, 1550668, 1550669, 1550670, 1550671, 1550672, 1550673, 1550674, 1550675, 1550676, 1550677, 1550678, 1550679, 1550680, 1550681, 1550682, 1550683, 1550684, 1550685, 1550686, 1550687, 1550688, 1550689, 1550690, 1550691, 1550692, 1550693, 1550694, 1550695, 1550696, 1550697, 1550698, 1550699, 1550700, 1550701, 1550702, 1550703, 1550704, 1550705, 1550706, 1550707)
  CACHE User Load (0.0ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` IN (8496, 455)
  Image Load (1.1ms)  SELECT `images`.* FROM `images` WHERE `images`.`id` IN (1550397, 1550664, 1550667, 1550668, 1550671, 1550672, 1550675, 1550676, 1550678, 1550680, 1550682, 1550684, 1550687, 1550688, 1550690, 1550692, 1550693, 1550695, 1550697, 1550699, 1550702, 1550703, 1550705, 1550706)
  ImageVote Load (0.9ms)  SELECT `image_votes`.* FROM `image_votes` WHERE `image_votes`.`image_id` IN (1550397, 1550664, 1550667, 1550668, 1550671, 1550672, 1550675, 1550676, 1550678, 1550680, 1550682, 1550684, 1550687, 1550688, 1550690, 1550692, 1550693, 1550695, 1550697, 1550699, 1550702, 1550703, 1550705, 1550706)
QUERY finished: model=Observation, flavor=all, params={:by=>"rss_log"}, time=0.771511751
  Rendering layout layouts/application.html.erb
  Rendering observations/index.html.erb within layouts/application
  Rendered application/content/_tab_set.html.erb (Duration: 1.9ms | Allocations: 240)
  Rendered application/content/_sorter.html.erb (Duration: 2.4ms | Allocations: 392)
   (262.4ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516551) AND (`votes`.`user_id` = 8496) AND (`votes`.`favorite` = TRUE)
  Vote Load (467.3ms)  SELECT `votes`.* FROM `votes` WHERE (`votes`.`observation_id` = 516551) AND (`votes`.`user_id` = 8496) AND (`votes`.`favorite` = TRUE) ORDER BY `votes`.`id` ASC LIMIT 1
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516551) AND (`votes`.`user_id` = 8496) AND (`votes`.`favorite` = TRUE)
  Vote Load (413.4ms)  SELECT `votes`.* FROM `votes` WHERE (`votes`.`observation_id` = 516551) AND (`votes`.`user_id` = 8496) AND (`votes`.`favorite` = TRUE)
  Naming Load (0.4ms)  SELECT `namings`.* FROM `namings` WHERE `namings`.`id` = 710892 LIMIT 1
  Name Load (0.6ms)  SELECT `names`.* FROM `names` WHERE `names`.`id` = 25237 LIMIT 1
   (270.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516707) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516707) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (286.1ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516706) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (285.5ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516705) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516705) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (281.4ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516704) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516704) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (278.8ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516703) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516703) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (273.5ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516702) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516702) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (278.6ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516701) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516701) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (276.2ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516700) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516700) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (287.5ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516699) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (271.2ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516698) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516698) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (274.7ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516697) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516697) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (279.8ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516696) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516696) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (278.5ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516695) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516695) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (279.4ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516694) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516694) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (280.6ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516693) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516693) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (278.2ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516692) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516692) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (275.9ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516691) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516691) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (275.8ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516690) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516690) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (274.5ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516689) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516689) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (283.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516688) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516688) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (278.4ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516687) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516687) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (264.2ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516686) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516686) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
   (280.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516685) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
  CACHE  (0.0ms)  SELECT COUNT(*) FROM `votes` WHERE (`votes`.`observation_id` = 516685) AND (`votes`.`user_id` = 455) AND (`votes`.`favorite` = TRUE)
```

Reply:
Intriguing side-note from my reading at the moment — Turbo and Stimulus offer a way to lazy-load the image caption text from the `ImagesController#show` action `format:turbo_stream` response into a blank lightbox caption, when the lightbox hits that image. That would eliminate having to render the captions ahead of time and get rid of a lot of these includes.

The same is possible for the image voting interface, it can be lazy-loaded too.

Reply:
I assume that lazy-loading still means separate queries for each image, only it does it later, right?  But presumably it may never need to if a user doesn't flip through most of the carousels?  It's a tricky call.  The user experience will include long delays as it loads individual image information.  Or even if it lazy-loads the info for all of the images for one carousel at once, there will still be at least one few-second delay each time the user goes to look at images in a carousel?  That figure -- "few-second" -- may be my unique experience, though, based on poor internet connectivity, and not a fair characterization of most users' experience?

Reply:
Going to mark the original issue as resolved. The query weirdness was a separate problem arising from the `show_obs_title`, fixed this week.

In this case lazy-loading does mean that if the user doesn't flip through the carousel, they won't load the image.